scripts/article1_post_draw.gml

```diff
@@ -38,110 +45,98 @@ if (x_dist == 0) {
     start_x = left_x;
     target_arc_length = chain_length / chain_segments;
 
+    end_iter = chain_segments - 1;
```

Comment:
to fix the jank, we now render the chain from left to right until we pass the vertex, then we render from right to left. end_iter is initially set to the max amount of chain segments so that we skip the right side pass if all segments are to the left of the vertex (this is possible in cases where the left character is much higher than they are outward from the other character. when we do the backward pass at line 71 we start at the max chain segments and decrement until we reach end iter.